src/common.rs

```diff
@@ -13,11 +15,17 @@ pub struct SourcedPackage {
 }
 
 pub fn sourced_dependencies(extra_options: Vec<String>) -> Vec<SourcedPackage> {
-    let meta = MetadataCommand::new()
+    let meta = match MetadataCommand::new()
         .features(AllFeatures)
         .other_options(extra_options)
         .exec()
-        .unwrap();
+    {
+        Ok(v) => v,
+        Err(cargo_metadata::Error::CargoMetadata { stderr: _ }) => err_exit(
+            "Could not find Cargo.toml in current or parent directory, make sure your inside of a crate!",
+        ),
+        Err(_) => err_exit("Unknown error whilst fetching crate's metadata!"),
```

Comment:
Why are we using custom error messages here? `MetadataCommand::new().exec()` already provides perfectly good error messages, I think we can just print them without any matching. I don't want to destroy the informative error messages we get from `cargo metadata` and show only "unknown error" which is not actionable for the user.

For reference, this is the output I get without this PR:

thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: CargoMetadata { stderr: "error: could not find `Cargo.toml` in `/home/shnatsel` or any parent directory\n" }', /home/shnatsel/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-supply-chain-0.1.2/src/common.rs:20:10